src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -544,6 +544,16 @@ void os::print_context(outputStream *st, const void *context) {
   st->print(", ERR=" INTPTR_FORMAT, (intptr_t)uc->uc_mcontext.gregs[REG_ERR]);
   st->cr();
   st->print("  TRAPNO=" INTPTR_FORMAT, (intptr_t)uc->uc_mcontext.gregs[REG_TRAPNO]);
+#ifndef MUSL_LIBC
+  // Add XMM registers + MXCSR. Note that C2 uses XMM to spill GPR values including pointers.
+  st->cr();
+  st->cr();
+  for (int i = 0; i < 16; ++i) {
```

Comment:
Better check if `fpregs` points into `*uc` before dereferencing it (asserting that it actually does).

Reply:
An assertion is not ideal. We don't want to run into it while printing an hs_err file. I've added a sanity check and print another message instead if it's bad.